### PR TITLE
Calculate the idx of states and state_choices per period ex ante

### PR DESCRIPTION
**Pull request by MaxBlesch**

### CODE REVIEW

File: src/dcegm/solve.py [hunk 1]
```diff
@@ -109,8 +110,18 @@ def solve_dcegm(
         compute_utility=compute_utility,
         compute_marginal_utility=compute_marginal_utility,
     )
+    (
+        state_choices_per_period,
+        states_per_period,
+    ) = determine_states_and_state_choices_per_period(
```
Comment (segsell) [inline src/dcegm/solve.py]: Can you rename that to ```determine_states_and_state_choice_combs_per_period```?
Then it's consistent with our variable naming.

File: src/dcegm/solve.py [hunk 2]
```diff
@@ -109,8 +110,18 @@ def solve_dcegm(
...
+        state_space=state_space,
+        state_choice_space=state_choice_space,
+        num_periods=n_periods,
+    )
 
     backwards_induction(
+        state_choices_per_period=state_choices_per_period,
```
Comment (segsell) [inline src/dcegm/solve.py]: state_choice_combs_per_period

File: src/dcegm/state_space.py [hunk 3]
```diff
@@ -264,3 +270,36 @@ def create_current_state_and_state_choice_objects(
         reshape_current_state_choice_vec_to_mat,
         transform_between_state_and_state_choice_vec,
     )
+
+
+def determine_states_and_state_choices_per_period(
+    state_space, state_choice_space, num_periods
+):
+    """Determine states and state choices per period and save as dictionaries.
```
Comment (segsell) [inline src/dcegm/state_space.py]: Adjust docstrings: Create dicts of period-specific states and state-choice combinations.

**Review by segsell — CHANGES_REQUESTED** [review]

File: src/dcegm/solve.py (hunk 2 above)
Comment (MaxBlesch) [reply src/dcegm/solve.py]: Let us abandon combs in future. They are state choices

File: src/dcegm/state_space.py (hunk 3 above)
Comment (MaxBlesch) [reply src/dcegm/state_space.py]: Adjusted docstring!

File: src/dcegm/solve.py (hunk 1 above)
Comment (MaxBlesch) [reply src/dcegm/solve.py]: Other function name, but lets abandon combs!

**Review by segsell — APPROVED** [review]

**Pull request merged**